add kapa 1.6.2 release notes (#3164)
influxdata/docs-v2: content/kapacitor/v1.6/about_the_project/releasenotes-changelog.md

    name: Release notes
---

## v1.6.2 [2021-08-04]

### Features

> - Add the `-template-id` option to `task list`.

Author: Does this refer to the `kapacitor flux task list`?
https://docs.influxdata.com/kapacitor/v1.6/working/flux/manage/list/

Reviewer: Is this for `feat(tasks): add template-id to task list (#2621)` ? This is about the swagger API change to add data to the response, not a new option.

Author: Thank you @lesam!  I will swap the term "option" for "property".

    name: Release notes
---

## v1.6.2 [2021-09-24]

### Features

> - Add the `-template-id` property to the `GET /kapacitor/v1/tasks` request response. Adding this property helps to identify tasks that were created from a template.

Reviewer: `s/-template-id/template-id` I think

Author: @lesam do you mean replace `-template-id` with `s/-template-id/template-id`?

Author: when you send the `GET /kapacitor/v1/tasks` requests, the `template-id` is included in the response

Reviewer: Sorry, I meant replace `-template-id` with `template-id`.

- For alert templates, row templates, and details templates, add support for third-party services that reject standard `json` (with carriage return at end) by adding the ability to use compact `json` in templates. (Also compacts Big Panda alert details to avoid Panda service error.) jsonCompact.

### Bug fixes

- Previously, errors could occur if a measurement string had special characters. Now, `expvar` string json encoding handles special characters, thanks @prashanthjbabu!
- Update `jwt` dependencies of libraries to prevent this vulnerability: https://nvd.nist.gov/vuln/detail/CVE-2020-26160.
- Switch to `github.com/golang-jwt/jwt` to prevent this vulnerability: https://nvd.nist.gov/vuln/detail/CVE-2020-26160.
- Switch task service to use Flux formatter that preserves comments.

> - Previously, Kapacitor used a remote `influxdb` bucket (or 1.x database) to store Flux task run logs, and you had to manually create the database or bucket to store Flux logs. Now, Kapacitor automatically creates an InfluxDB 1.x database or InfluxDB 2.x bucket for Flux task logs. Note for InfluxDB 1.x, the `database` bucket name is created (rather than the 1.x format `database/rp` that includes the retention policy.

Reviewer: This one is probably a feature rather than a fix (sorry if it was in the wrong place in the original kapa changelog!)

Reviewer: Also is there a PR for this yet? https://github.com/influxdata/docs-v2/issues/3131

Author: thank you @lesam, looks like it moved to feature in change log, will update!

Author: updated PR to include https://github.com/influxdata/docs-v2/issues/3131

    name: Release notes
---

## v1.6.2 [2021-09-24]

### Features

> - Add the `-template-id` property to the `GET /kapacitor/v1/tasks` request response. Adding this property helps to identify tasks that were created from a [template](/kapacitor/v1.6/working/template_tasks/).

Reviewer 2: ...oh, just realized `-template-id` is probably the CLI option. Maybe we should distinguish between CLI and API usage of template-id?

> - Add the `-template-id` property to the `GET /kapacitor/v1/tasks` request response. Adding this property helps to identify tasks that were created from a [template](/kapacitor/v1.6/working/template_tasks/).

Author: @jstirnaman, yes, was unintentional, the team clarified that no CLI changes were made for this release

- For alert templates, row templates, and details templates, add support for third-party services that reject standard `json` (with carriage return at end) by compacting `json` in templates. To do this, replace `{{ json . }}` with `{{ jsonCompact . }}` in your templates. (This change also compacts Big Panda alert details to avoid Panda service error.)

> - Previously, Kapacitor used a remote `influxdb` bucket (or 1.x database) to store Flux task run logs, and you had to manually create the database or bucket to store Flux logs. Now, Kapacitor automatically creates an InfluxDB 1.x database or InfluxDB 2.x bucket for Flux task logs (by default, `kapacitor_fluxtask_logs`). Note for InfluxDB 1.x, the `database` bucket name is created (rather than the 1.x format `database/rp` that includes the retention policy. For more information, see how to [use Flux tasks with Kapacitor](/kapacitor/v1.6/working/flux/).

Reviewer: I would just put:

```
- Previously, Kapacitor used a remote `influxdb` bucket (or 1.x database) to store Flux task run logs, and you had to manually create the database or bucket to store Flux logs. Now, Kapacitor automatically creates an InfluxDB 1.x database or InfluxDB 2.x bucket for Flux task logs (by default, `kapacitor_fluxtask_logs`). For more information, see how to [use Flux tasks with Kapacitor](/kapacitor/v1.6/working/flux/).
```

### Bug fixes

> - Previously, errors could occur if a measurement string had special characters. Now, `expvar` string json encoding handles special characters, thanks @prashanthjbabu!

Author: , thanks @prashanthjbabu

> - Previously, if `disable-subscriptions=true` in the [InfluxDB](/kapacitor/v1.6/administration/configuration/#influxdb) section of the Kapacitor configuration file, Kapacitor didn't validate that the `influxdb` instance was running and available. Now, Kapacitor correctly validates (via `GET /ping`) whenever `enabled=true`. If InfluxDB is not available, Kapacitor does not start.

Reviewer 3: ```suggestion
- Correctly validate that connected InfluxDB instances are running when`disable-subscriptions` is set to `true` in the [InfluxDB section of the Kapacitor configuration file](/kapacitor/v1.6/administration/configuration/#influxdb). If InfluxDB is not available, Kapacitor does not start.
```

> - Update `jwt` dependencies of libraries to prevent this vulnerability: https://nvd.nist.gov/vuln/detail/CVE-2020-26160.
> - Switch to `github.com/golang-jwt/jwt` to prevent this vulnerability: https://nvd.nist.gov/vuln/detail/CVE-2020-26160.

Reviewer 3: ```suggestion
- Update `jwt` dependencies and switch to `github.com/golang-jwt/jwt` to remediate the [CVE-2020-26160 vulnerability](https://nvd.nist.gov/vuln/detail/CVE-2020-26160).
```
